### ISSUE TITLE
Review Conda Channel Policies

### ISSUE THREAD

**Issue by ZLLentz** [pcdshub/pcds-envs#119]
Status quo:
1. All packages upload to `pcds-dev` via TravisCI when we merge a pull request into master
2. All packages upload to `pcds-tag` via TravisCI when we make a tag on master
3. Some packages upload to `conda-forge` using the normal conda forge mechanisms
4. Create environments using channel priority `pcds-tag` > `conda-forge` > `defaults`
4b. For the special dev builds, add `pcds-dev` as the highest priority channel

This is a bit confusing and could use some consolidation.

**Comment by ZLLentz** [comment]
@klauer @hhslepicka @ZryletTC I'm starting to think we need to reorganize because having three channels is getting unwieldy. let me know what you think.

**Comment by hhslepicka** [comment]
I vote to deprecate pcds-tag in favor of conda-forge if we can put it all there and leave pcds-dev as a good dev area.

**Comment by ZLLentz** [comment]
Where do we put random extra builds we need if we deprecate pcds-tag?

**Comment by hhslepicka** [comment]
Which packages would fit in this category?

**Comment by ZLLentz** [comment]
anything we slapped into the pcds-recipes repo at one time or another if we need to bring it back

**Comment by ZLLentz** [comment]
I think those are all on conda-forge now... maybe it is a contingency that doesn't really exist

**Comment by ZLLentz** [comment]
If there is a consensus on this then I'm ok with using only `conda-forge`, but I refuse to maintain any recipe for packages that I think should be eliminated or merged into other packages... which are the following:
`hxrsnd` (should be merged into `pcdsdevices`),
`pswalker` (should be renamed and made more general... Teddy and I were planning to rewrite the whole thing as `nabs` for l2si automation),
`transfocate` (should be merged into `pcdsdevices`),

**Comment by ZLLentz** [comment]
We also have a few lingering small projects on `pcds-tag` that can probably continue to hang out there without bothering anyone.

**Comment by hhslepicka** [comment]
I can't say about consensus as this is a PCDS/ECS decision to be made... I think that it is not a big issue to keep pcds-tag channel for those packages and for historic reasons.

**Comment by ZLLentz** [comment]
I think it is at least uncontroversial to stop doing pcds-tag builds for things on conda-forge and to speed up the tests by using pip where possible?

**Comment by hhslepicka** [comment]
Do you mean pip to test with latest of all packages?

**Comment by ZLLentz** [comment]
Most of our travis builds take 90% of their time just doing `conda build` and `conda install`, when they could be doing it like this using pcds-ci-helpers https://github.com/pcdshub/pcds-ci-helpers/blob/master/example_python_travis.yml

**Comment by hhslepicka** [comment]
That is only because we waste too much time doing OS specific builds.
I was able to bypass this issue by using `noarch` package and saving the build package as an artifact to be used by the test steps.

Here is an example: https://dev.azure.com/pydm/pydm/_build/results?buildId=389&view=results
It takes 5 minutes to build and another 5 to run 10 tests in parallel.
If we can find a way to do the same with Travis that would save a lot of time.
The packages at conda-forge are already `noarch` so it is just a matter of copying the recipes to our repos as well.

If we use pip to test we are subject to difference at system packages too. How to deal with that? Other than that I am fine with it.

**Comment by ZLLentz** [comment]
Is it worth it to find ways to do this with Travis rather than just switching to Azure? You're making it sound like we need to proceed slowly and deliberately with stuff like this to make sure we get the best outcome.

**Comment by ZLLentz** [comment]
I'm also worried about the conda-forge recipe and the in-repo recipe diverging. Perhaps it's possible to just clone the feedstock during the tests if a build needs to be done?

**Comment by hhslepicka** [comment]
I moved to Azure just because I had the need to run more jobs in parallel and test with Linux, Windows and macOS... otherwise I would be happy with Travis.

I do like the fact that with Azure I can upload artifacts and do some other nice things like using yaml templates (Travis allow that now, not as good as Azure yet but they are developing this feature more).

About the recipes getting out of sync that is a risk but the feedstock uses a tagged version and for the tests we want to build with the local code. We would need to overcome that.

**Comment by ZLLentz** [comment]
Fair enough. I think I will push for just a separate build matrix task that checks the conda-forge build against the master's test suite so we have a warning of when the conda-forge recipe needs to be updated.
